python/examples/tabulated.py

```diff
@@ -0,0 +1,156 @@
+z_ln = Jn_zeros(lmax, nmax)
+z_nl = z_ln.T
+
+
+def R_nl(n, el, r):
+    # Un-normalized LE radial basis functions
+    return j_l(el, z_nl[n, el] * r / a)
+
+
+def N_nl(n, el):
+    # Normalization factor for LE basis functions, excluding the a**(-1.5) factor
+    def function_to_integrate_to_get_normalization_factor(x):
+        return j_l(el, x) ** 2 * x**2
+
+    integral, _ = sp.integrate.quadrature(
+        function_to_integrate_to_get_normalization_factor, 0.0, z_nl[n, el]
+    )
+    return (1.0 / z_nl[n, el] ** 3 * integral) ** (-0.5)
+
+
+def get_LE_function(n, el, r):
+    R = np.zeros_like(r)
+    for i in range(r.shape[0]):
+        R[i] = R_nl(n, el, r[i])
+    return N_nl(n, el) * R * a ** (-1.5)
+
+
+# Quick normalization check:
+def random_LE_function(x):
+    return get_LE_function(1, 1, x) ** 2 * x**2
+
+
+normalization_check_integral, _ = sp.integrate.quadrature(random_LE_function, 0.0, a)
+print(f"Normalization check (needs to be close to 1): {normalization_check_integral}")
+
+
+# Now the derivatives (by finite differences):
+def get_LE_function_derivative(n, el, r):
+    delta = 1e-6
+    all_derivatives_except_at_zero = (
+        get_LE_function(n, el, r[1:] + delta) - get_LE_function(n, el, r[1:] - delta)
+    ) / (2.0 * delta)
+    derivative_at_zero = (
+        get_LE_function(n, el, np.array([delta / 10.0]))
+        - get_LE_function(n, el, np.array([0.0]))
+    ) / (delta / 10.0)
+    return np.concatenate([derivative_at_zero, all_derivatives_except_at_zero])
+
+
+# The radial basis functions and their derivatives can be input into a spline
+# generator. This will output the positions of the spline points, the
+# values of the basis functions evaluated at the spline points, and the
+# corresponding derivatives.
+positions, values, derivatives = generate_splines(
+    get_LE_function, get_LE_function_derivative, nmax, lmax, a
+)
+
+# Feed the splines to the Rust calculator:
+# (IMPORTANT: "atomic_gaussian_width" and "cutoff_function" will be ignored)
+
+hypers_spherical_expansion = {
+    "cutoff": a,
+    "max_radial": nmax,
+    "max_angular": lmax,
+    "center_atom_weight": 0.0,
+    "radial_basis": {
+        "Tabulated": {
+            "positions": positions,
+            "values": values,
+            "derivatives": derivatives,
+        }
+    },
+    "atomic_gaussian_width": 1.0,  # ignored
+    "cutoff_function": {"Step": {}},  # ignored
+}
+calculator = SphericalExpansion(**hypers_spherical_expansion)
+
+# Create dummy structures to test if the calculator outputs correct radial functions:
+
+
+def get_dummy_structures(r_array):
+    dummy_structures = []
+    for r in r_array:
+        dummy_structures.append(ase.Atoms("CH", positions=[(0, 0, 0), (0, 0, r)]))
+    return dummy_structures
+
+
+r = np.linspace(0.1, 4.9, 20)
+structures = get_dummy_structures(r)
+spherical_expansion_coefficients = calculator.compute(structures)
+
+# Extract l = 0 features and check that the n = 2 predictions are the same:
```

Comment:
why only the n=2?

Reply:
Just a random check. We can add more n values. It would be more difficult to do l>0 checks because one would need to divide by the spherical harmonics, which are not constant anymore in that case